server/leafnode_test.go

```diff
@@ -4072,3 +4099,942 @@ func TestLeafNodeInterestPropagationDaisychain(t *testing.T) {
+}
+`
+	akp, err := nkeys.CreateAccount()
+	require_NoError(t, err)
+	creds := createUserWithLimit(t, akp, time.Time{}, func(pl *jwt.UserPermissionLimits) {
+		pl.Pub.Deny.Add(jsAllAPI)
+		pl.Sub.Deny.Add(jsAllAPI)
+	})
+
+	sd := createDir(t, JetStreamStoreDir)
+	defer os.RemoveAll(sd)
+
+	confL := createConfFile(t, []byte(fmt.Sprintf(tmplL, sd, creds, creds)))
+	defer removeFile(t, confL)
+	opts := LoadConfig(confL)
+	sL, err := NewServer(opts)
+	require_NoError(t, err)
+
+	l := captureNoticeLogger{}
+	sL.SetLogger(&l, false, false)
+
+	go sL.Start()
+	defer sL.Shutdown()
+
+	// wait till the notices got printed
+UNTIL_READY:
+	for {
+		<-time.After(50 * time.Millisecond)
+		l.Lock()
+		for _, n := range l.notices {
+			if strings.Contains(n, "Server is ready") {
+				l.Unlock()
+				break UNTIL_READY
+			}
+		}
+		l.Unlock()
+	}
+
+	l.Lock()
+	cnt := 0
+	for _, n := range l.notices {
+		if strings.Contains(n, "LeafNode Remote for Account A uses credentials file") ||
+			strings.Contains(n, "LeafNode Remote for System Account uses") ||
+			strings.Contains(n, "Remote for System Account uses restricted export permissions") ||
+			strings.Contains(n, "Remote for System Account uses restricted import permissions") {
+			cnt++
+		}
+	}
+	l.Unlock()
+	require_True(t, cnt == 4)
+}
+
+func TestLeafNodeJetStreamDefaultDomainCfg(t *testing.T) {
+	tmplHub := `
+listen: 127.0.0.1:%d
+accounts :{
+    A:{ jetstream: %s, users:[ {user:a1,password:a1}]},
+    SYS:{ users:[ {user:s1,password:s1}]},
+}
+system_account: SYS
+jetstream : %s
+server_name: HUB
+leafnodes: {
+	listen: 127.0.0.1:%d
+}
+%s
+`
+
+	tmplL := `
+listen: 127.0.0.1:-1
+accounts :{
+    A:{   jetstream: enable, users:[ {user:a1,password:a1}]},
+    SYS:{ users:[ {user:s1,password:s1}]},
+}
+system_account: SYS
+jetstream: { domain: "%s", store_dir: "%s", max_mem: 100Mb, max_file: 100Mb }
+server_name: LEAF
+leafnodes: {
+    remotes:[{url:nats://a1:a1@127.0.0.1:%d, account: A},%s]
+}
+%s
+`
+
+	test := func(domain string, sysShared bool) {
+		confHub := createConfFile(t, []byte(fmt.Sprintf(tmplHub, -1, "disabled", "disabled", -1, "")))
+		defer removeFile(t, confHub)
+		sHub, _ := RunServerWithConfig(confHub)
+		defer sHub.Shutdown()
+
+		noDomainFix := ""
+		if domain == _EMPTY_ {
+			noDomainFix = `default_js_domain:{A:""}`
+		}
+
+		sys := ""
+		if sysShared {
+			sys = fmt.Sprintf(`{url:nats://s1:s1@127.0.0.1:%d, account: SYS}`, sHub.opts.LeafNode.Port)
+		}
+
+		sdLeaf := createDir(t, JetStreamStoreDir)
+		defer os.RemoveAll(sdLeaf)
+		confL := createConfFile(t, []byte(fmt.Sprintf(tmplL, domain, sdLeaf, sHub.opts.LeafNode.Port, sys, noDomainFix)))
+		defer removeFile(t, confL)
+		sLeaf, _ := RunServerWithConfig(confL)
+		defer sLeaf.Shutdown()
+
+		lnCnt := 1
+		if sysShared {
+			lnCnt++
+		}
+
+		checkLeafNodeConnectedCount(t, sHub, lnCnt)
+		checkLeafNodeConnectedCount(t, sLeaf, lnCnt)
+
+		ncA := natsConnect(t, fmt.Sprintf("nats://a1:a1@127.0.0.1:%d", sHub.opts.Port))
```

Comment:
I think `defer ncA.Close()` is missing here.

Reply:
fixed